app/src/main/java/com/example/mobile/monitors/IMonitor.kt

```diff
@@ -4,5 +4,5 @@ interface IMonitor {
     //TODO: maybe also add pause/resume functionality   
     fun startMonitoring(onStart: () -> Unit)
     fun stopMonitoring()
-    fun readValue(): Double
+    suspend fun readValue(): Double

```

Comment:
perché queste sono `suspend` ora? se è perché vengono chiamate in una coroutine, credo che `suspend` vada spostato sui `MonitoringScreen`, se è per qualche requisito della libreria dei database può essere utile lasciare un commento per ricordarcelo